src/expm.jl

```diff
@@ -0,0 +1,64 @@
+@inline expm(A::StaticMatrix) = _expm(Size(A), A)
+
+@generated function _expm(::Size{(1,1)}, A::StaticMatrix)
+    @assert size(A) == (1,1)
+    T = promote_type(typeof(sqrt(one(eltype(A)))), Float32)
+    newtype = similar_type(A,T)
+
+    quote
+        $(Expr(:meta, :inline))
+        ($newtype)((exp(A[1]), ))
+    end
+end
+
+@generated function _expm{S<:Real}(::Size{(2,2)}, A::StaticMatrix{S})
+    @assert size(A) == (2,2)
+    T = promote_type(typeof(sqrt(one(eltype(A)))), Float32)
+    newtype = similar_type(A,T)
+
+    quote
+        $(Expr(:meta, :inline))
+
+        @inbounds a = A[1]
+        @inbounds c = A[2]
+        @inbounds b = A[3]
+        @inbounds d = A[4]
+
+        v = (a-d)^2 + 4*b*c
+
+        # if v == 0
+        z1::$T = 1.0
```

Comment:
The reason for this was to ensure type stability. Moved it to the else-block instead.